pkg/signature/kms/aws/signer.go

```diff
@@ -0,0 +1,254 @@
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return a.client.sign(ctx, digest, hf)
+}
+
+// PublicKey returns the public key that can be used to verify signatures created by
+// this signer. If the caller wishes to specify the context to use to obtain
+// the public key, pass option.WithContext(desiredCtx).
+//
+// All other options are ignored if specified.
+func (a *SignerVerifier) PublicKey(opts ...signature.PublicKeyOption) (crypto.PublicKey, error) {
+	ctx := context.Background()
+	for _, opt := range opts {
+		opt.ApplyContext(&ctx)
+	}
+
+	return a.client.public(ctx)
+}
+
+// VerifySignature verifies the signature for the given message. Unless provided
+// in an option, the digest of the message will be computed using the hash function specified
+// when the SignerVerifier was created.
+//
+// This function returns nil if the verification succeeded, and an error message otherwise.
+//
+// This function recognizes the following Options listed in order of preference:
+//
+// - WithDigest()
+//
+// - WithCryptoSignerOpts()
+//
+// - WithContext()
+//
+// All other options are ignored if specified.
+func (a *SignerVerifier) VerifySignature(sig, message io.Reader, opts ...signature.VerifyOption) (err error) {
+	ctx := context.Background()
+
+	for _, opt := range opts {
+		opt.ApplyContext(&ctx)
+	}
+
+	return a.client.verify(ctx, sig, message, opts...)
+}
+
+// VerifySignatureRemotely verifies the signature for the given message via KMS
+// API call. Unless provided in an option, the digest of the message will be
+// computed using the hash function specified when the SignerVerifier was created.
+//
+// This function returns nil if the verification succeeded, and an error message otherwise.
+//
+// This function recognizes the following Options listed in order of preference:
+//
+// - WithDigest()
+//
+// - WithCryptoSignerOpts()
+//
+// - WithContext()
+//
+// All other options are ignored if specified.
+func (a *SignerVerifier) VerifySignatureRemotely(sig, message io.Reader, opts ...signature.VerifyOption) (err error) {
+	var digest []byte
+	ctx := context.Background()
+
+	for _, opt := range opts {
+		opt.ApplyDigest(&digest)
+		opt.ApplyContext(&ctx)
+	}
+
+	var signerOpts crypto.SignerOpts
+	signerOpts, err = a.client.getHashFunc(ctx)
+	for _, opt := range opts {
+		opt.ApplyCryptoSignerOpts(&signerOpts)
+	}
+	hf := signerOpts.HashFunc()
+
+	if len(digest) == 0 {
+		digest, hf, err = signature.ComputeDigestForVerifying(message, hf, awsSupportedHashFuncs, opts...)
+		if err != nil {
+			return err
+		}
+	}
+
+	sigBytes, err := io.ReadAll(sig)
+	if err != nil {
+		err = errors.Wrap(err, "reading signature")
+	}
+	return a.client.verifyRemotely(ctx, sigBytes, digest)
+}
+
+// CreateKey attempts to create a new key in Vault with the specified algorithm.
+func (a *SignerVerifier) CreateKey(ctx context.Context, algorithm string) (crypto.PublicKey, error) {
+	return a.client.createKey(ctx, algorithm)
+}
+
+type cryptoSignerWrapper struct {
```

Comment:
This is duplicated from the other two KMS backends... should we factor this out into a different package?

Reply:
probably, but we could do that in a separate PR